common/metrics/tags.go

```diff
@@ -0,0 +1,29 @@
+package metrics
+
+const domainTag = "domain"
+
+// Tag is an interface to define metrics tags
+type Tag interface {
+	Key() string
+	Value() string
+}
+
+// DomainTag wraps a domain tag
+type DomainTag struct {
+	value string
+}
+
+// NewDomainTag returns a new domain tag
+func NewDomainTag(value string) DomainTag {
```

Comment:
Just a style preference. If you make DomainTag struct private (domainTag struct) - you could do something like following:

```
client.Scope(metrics.StartWFExecutionScope, 
metrics.DomainTag("domain-name"),
metrics.ShardIDTag(shardID)) 

reads better than 

client.Scope(metrics.StartWFExecutionScope, 
metrics.NewDomainTag("domain-name"),
metrics.NewShardIDTag(shardID))
```

Zap logger uses similar style for typed k/v tags

Reply:
Done.